e2e/hardhatTransactionFlow.spec.js

```diff
@@ -15,6 +18,40 @@ let account = null;
 const RAINBOW_WALLET_DOT_ETH = '0x7a3d05c70581bD345fe117c06e45f9669205384f';
 const TESTING_WALLET = '0x3Cb462CDC5F809aeD0558FBEe151eD5dC3D3f608';
 
+const ETH_ADDRESS = 'eth';
+const BAT_TOKEN_ADDRESS = '0x0d8775f648430679a709e98d2b0cb6250d2887ef';
+const CRYPTOKITTIES_ADDRESS = '0x06012c8cf97BEaD5deAe237070F9587f8E7A266d';
+
+const getOnchainBalance = async (address, tokenContractAddress) => {
+  const provider = new JsonRpcProvider(
+    device.getPlatform() === 'ios'
+      ? process.env.HARDHAT_URL_IOS
+      : process.env.HARDHAT_URL_ANDROID,
+    'any'
+  );
+  if (tokenContractAddress === ETH_ADDRESS) {
+    const balance = await provider.getBalance(RAINBOW_WALLET_DOT_ETH);
+    return balance;
+  }
+  if (tokenContractAddress === CRYPTOKITTIES_ADDRESS) {
+    const kittiesContract = new Contract(
+      CRYPTOKITTIES_ADDRESS,
+      kittiesABI,
+      provider
+    );
+    const ownerAddress = await kittiesContract.ownerOf('1368227');
+    return ownerAddress === address;
```

Comment:
It's kind of weird that everything returns a balance except here. I'd move it to a separate function isNFTOwner or something like that.